### PR TITLE
LPD-41227 Handle the cases where the recipient property is retrieved as Object[] instead of List<Map<String, String>>

### PR DESCRIPTION
**Pull request by MarinhoFeliphe**
Related: https://liferay.atlassian.net/browse/LPD-41227

## Summary

When the NotificationTemplate DTO is build by `notification.rest.resource` will retrieve an `Object[]` instead of a `List<Map<String, String>>` for the recipient `To` if this one has the `toType` as `role`

### CODE REVIEW

**Comment by MarinhoFeliphe** [comment]
ci:test:relevant

**Comment by MarinhoFeliphe** [comment]
ci:test:sf

File: modules/apps/notification/notification-rest-test/src/testIntegration/java/com/liferay/notification/rest/resource/v1_0/test/NotificationTemplateResourceTest.java [hunk 1]
```diff
@@ -188,6 +191,52 @@ public void testPostNotificationTemplate() throws Exception {
...
+			"subject", JSONUtil.put("en_US", RandomTestUtil.randomString())
+		).put(
+			"type", NotificationConstants.TYPE_EMAIL
+		).toString();
+
+		NotificationTemplateResource.Builder
+			notificationTemplateResourceBuilder =
+				_notificationTemplateResourceFactory.create();
+
+		NotificationTemplateResource restAPINotificationTemplateResource =
```
Comment (linolaoj) [inline modules/apps/notification/notification-rest-test/src/testIntegration/java/com/liferay/notification/rest/resource/v1_0/test/NotificationTemplateResourceTest.java:229]: Hi @MarinhoFeliphe, can it be renamed to `restNotificationTemplateResource`?

**Review by linolaoj — CHANGES_REQUESTED** [review]

**Comment by MarinhoFeliphe** [comment]
Resent here: https://github.com/brianchandotcom/liferay-portal/pull/155960

**Pull request closed without merging**